sdl_viewer/src/graphic/tiled_texture_loader.rs

```diff
@@ -0,0 +1,107 @@
+use byteorder::{LittleEndian, ReadBytesExt};
+use image::{GenericImage, GenericImageView, ImageBuffer, LumaA, Pixel, Rgba};
+use num_integer::Integer;
+use std::collections::HashMap;
+use std::fs::File;
+use std::io::BufReader;
+use std::path::Path;
+
+/// Little trait that just dispatches to the correct read_xyz_into call for the image type
+pub trait ReadLittleEndian {
+    fn read_from<R: ReadBytesExt>(&mut self, rdr: &mut R) -> Result<(), std::io::Error>;
+}
+
+impl ReadLittleEndian for ImageBuffer<LumaA<f32>, Vec<f32>> {
+    fn read_from<R: ReadBytesExt>(&mut self, rdr: &mut R) -> Result<(), std::io::Error> {
+        rdr.read_f32_into::<LittleEndian>(self)
+    }
+}
+
+impl ReadLittleEndian for ImageBuffer<Rgba<u8>, Vec<u8>> {
+    fn read_from<R: ReadBytesExt>(&mut self, rdr: &mut R) -> Result<(), std::io::Error> {
+        rdr.read_exact(self)
+    }
+}
+
+type TilePos = (i32, i32);
+
+pub struct TiledTextureLoader<P: Pixel> {
+    tile_size: u32,
+    tiles: HashMap<TilePos, ImageBuffer<P, Vec<P::Subpixel>>>,
+}
+
+impl<P> TiledTextureLoader<P>
+where
+    P: Pixel + 'static,
+    ImageBuffer<P, Vec<P::Subpixel>>: ReadLittleEndian,
+{
+    pub fn new<I, Q>(tile_size: u32, tile_iter: I) -> Result<Self, std::io::Error>
+    where
+        I: Iterator<Item = ((i32, i32), Q)>,
```

Comment:
nit: use `TilePos` instead of `(i32, i32)`?

Reply:
Yep